### DISCUSSION TITLE
status code 400: Files are empty

### DISCUSSION THREAD

**Discussion by hkovacs** [strapi/strapi#5841] [Legacy Topics]
**Describe the bug**
I have implemented file uploading to api from frontend using the guide here https://strapi.io/documentation/3.0.0-beta.x/plugins/upload.html#upload-file-during-entry-creation

I get this response in my network:
```
{"statusCode":400,"error":"Bad Request","message":[{"messages":[{"id":"Upload.status.empty","message":"Files are empty"}]}]}
```

the network headers form data:
```
files.file: (binary)
data: {"field":"notices","title":"asdfsf","refId":"5da979e69bd03850f90e35b3"}
```

the network headers form data source:
```
------WebKitFormBoundaryKnEib84VcASIFIwj
Content-Disposition: form-data; name="files.file"; filename="a-test-file.txt"
Content-Type: text/plain


------WebKitFormBoundaryKnEib84VcASIFIwj
Content-Disposition: form-data; name="data"

{"field":"notices","title":"asdfsf","refId":"5da979e69bd03850f90e35b3"}
------WebKitFormBoundaryKnEib84VcASIFIwj--
```

**Steps to reproduce the behavior**
see my code below

**Expected behavior**
i expect a successful result as i am following the guide.

**Screenshots**
none

**Code snippets**
```
        const request = new XMLHttpRequest();

        const formData = new FormData();
        const formElement = document.querySelector('#upload_form');
        const formElements = formElement.elements;
        const data = {};

        for (let i = 0; i < formElements.length; i++) {
          const currentElement = formElements[i];
          if (!['submit', 'file'].includes(currentElement.type)) {
            data[currentElement.name] = currentElement.value;
          } else if (currentElement.type === 'file') {
            if (currentElement.files.length === 1) {
              const file = currentElement.files[0];
              formData.append(`files.${currentElement.name}`, file, file.name);
            } else {
              for (let i = 0; i < currentElement.files.length; i++) {
                const file = currentElement.files[i];
                formData.append(`files.${currentElement.name}`, file, file.name);
              }
            }
          }
        }
        formData.append('data', JSON.stringify(data));

        request.open('POST', `${process.env.REACT_APP_API_ENDPOINT}/upload`);
        request.setRequestHeader('Authorization', jwt);
        request.send(formData);
```
**System**

- Node.js version: v10.16.3
- NPM version: 6.9.0
- Strapi version: Strapi v3.0.0-beta.18.3
- Database: mongo v4.0.12
- Operating system: ubuntu 18.04

**Additional context**
I have not had time to update to 19 or 20, but there should not be a need to as the guide has been the same since before 18.

I am happy to update to 19 or 20 if there is a fix in that version specific to this issue. my search did not find a fix in recent ticket...

thank you

**Comment by derrickmehaffy** [comment]
Please test on the latest version, you may not need to update but at least test with a fresh project on the latest version.

(Guide or not, bug reports must be for the latest version regardless)

**Comment by hkovacs** [comment]
roger that. i will do the update as it is the better time investment.

**Comment by hkovacs** [comment]
upgraded to beta20. everything works!

suffice it to say, i hesitate to upgrade in a timely fashion as previous upgrades have been incompletely documented and releases have been buggy.

beta 20 fixed a bunch of bugs.

kudos!
